assets/src/apps/authoring/components/Modal/diagnostics/DiagnosticMessage.tsx

```diff
@@ -35,6 +35,26 @@ export const ValueUndefined: React.FC<Message> = ({ problem }: Message) => (
   </span>
 );
 
+export const InvalidExpressionValue: React.FC<Message> = ({ problem }: Message) => (
+  <span>
+    The &quot;
+    <strong>{problem?.item?.rule?.name}</strong>&quot; rule has invalid expression.
+  </span>
+);
+
+export const InvalidPartExpressionValue: React.FC<Message> = ({ problem }: Message) => (
+  <span>
+    The &quot;
+    <strong>{problem?.item?.part?.id}</strong>&quot; component has invalid expression in &nbsp;
+    {problem?.item?.part?.custom?.configData && (
+      <span>
+        configData - &quot;<strong>{problem?.item?.key}</strong>&quot; variable
+      </span>
+    )}
+    {problem?.item?.part?.custom?.mcqItems && <span> MCQ Options</span>}
```

Comment:
hmm it would be good if we don't need part specific things here, however maybe we can leave it as a TODO

Reply:
Done